caproto/_data.py

```diff
@@ -249,6 +271,7 @@ def get_dbr_data(self, type_):
     async def set_dbr_data(self, data, data_type, metadata):
         '''Set data from DBR metadata/values'''
         self.value = self.fromtype(values=data, data_type=data_type)
+        self.timestamp = time.time()
```

Comment:
I think we probably should just check if it's present in `metadata`, otherwise use `time.time()`?